source/style-guide/nested-components.txt

```diff
@@ -0,0 +1,372 @@
+.. _nested-components:
+
+============================
+Nested Components Guidelines
+============================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. default-domain:: mongodb
+
+To allow LLMs to consume our content easily, don't nest the following components:
+
+- Admonitions inside admonitions
+- Admonition inside tables
+- Examples inside admonitions
+ 
+Use the following guidance to fix existing published nested components 
+or replace nested components in your pull requests.
+
+Admonitions Inside Admonitions
+------------------------------
+
+Remove the nested admonition and make its contents part of the parent 
+admonition, or remove it from admonitions altogether. The text may 
+follow directly after the preceding text, or you may include a line 
+break if appropriate. If the nested admonition is an example, see 
+:ref:`examples-in-admonitions`. If the nested admonition is an 
+Important or Warning admonition, see :ref:`important-or-warning`
+in the :ref:`examples-in-admonitions` section.
+
+**Example:**
+
+.. list-table::
+   :header-rows: 1
+
+   * - Before
+     - After
+
+   * - 
+       .. important:: AWS Only
+
+          Amazon Web Services (AWS) Virtual Private Cloud (VPC) peering
+          connections are region-specific. Clusters utilizing an 
+          existing VPC peering connection to an AWS VPC in a given AWS 
+          region lose access to that peering connection if moved to a 
+          different AWS region. The moved cluster may use an existing 
+          peering connection in the new region.
+
+          .. seealso:: 
+   
+             :atlas:`Set Up a Network Peering Connection 
+             </security-vpc-peering>`. 
+     - 
+       .. important:: AWS Only
```

Comment:
Meta question. Since the "after" here is a nested component because it's a callout inside a table (even if the table is just for showing examples), do we think that might confuse people? It's probably fine, but just wondering. I think it was easier to not get confused when it was screenshots for the examples

Reply:
@sarahsimpers, I thought the same thing, but since the rendering is essentially the same for a screenshot versus the actual rST, I resigned myself to it. The alternative is to have Before and After sections. Let me know your thoughts.

Reply:
Yeah, let's leave it, I'm overthinking.